src/integrator.jl

```diff
@@ -1,9 +1,13 @@
 abstract type AbstractIntegrator end
 
-struct Leapfrog{T<:Real} <: AbstractIntegrator
+struct Leapfrog{T<:AbstractFloat} <: AbstractIntegrator
```

Comment:
I updated a few type constraints from `Real` to `AbstractFloat`. E.g. here step size can only be `AbstractFloat`

Reply:
@mohamed82008 Does this sound good?